docs/turbo-native-modules.md

```diff
@@ -0,0 +1,255 @@
+        backgroundColor={backgroundStyle.backgroundColor}
+      />
+
+      <Text style={styles.text}>
+        Current stored value is: {value ?? 'No Value'}
+      </Text>
+
+      <TextInput
+        placeholder="Enter the text you want to store"
+        style={styles.textInput}
+        onChangeText={setEditingValue}
+      />
+
+      <Button title="Save" onPress={saveValue} />
+      <Button title="Delete" onPress={deleteValue} />
+      <Button title="Clear" onPress={clearAll} />
+    </SafeAreaView>
+  );
+}
+
+const styles = StyleSheet.create({
+  text: {
+    margin: 10,
+    fontSize: 20,
+  },
+  textInput: {
+    margin: 10,
+    height: 40,
+    borderColor: 'black',
+    borderWidth: 1,
+    paddingLeft: 5,
+    paddingRight: 5,
+    borderRadius: 5,
+  },
+});
+
+export default App;
+```
+
+## 3. Configure Codegen to run
+
+The spec is used by the React Native Codegen tools to generate platform specific interfaces and boilerplate for us. To do this, Codegen needs to know where to find our spec and what to do with it. Update your package.json to include:
+
+```diff
+     "start": "react-native start",
+     "test": "jest"
+   },
+   // highlight-add-start
++  "codegenConfig": {
++    "name": "NativeLocalStorageSpec",
++    "type": "modules",
++    "jsSrcsDir": "specs",
++    "android": {
++      "javaPackageName": "com.nativelocalstorage"
++    }
++  },
+   // highlight-add-end
+   "dependencies": {
+     "react": "18.3.1",
+     "react-native": "0.75.3"
+```
+
+:::info
+Codegen is called at build-time, but you can hook it into your preferred build too if you’d like. See [using Codegen](the-new-architecture/using-codegen) our more detailed documentation.
+:::
+
+With everything wired up for Codegen, we need to prepare our native code to hook into our generated code.
+
+<Tabs groupId="platforms" queryString defaultValue={constants.defaultPlatform}>
+<TabItem value="android" label="Android">
+Codegen is executed through the `generateCodegenArtifactsFromSchema` Gradle task:
+
+```bash
+cd android
+./gradlew generateCodegenArtifactsFromSchema
+
+BUILD SUCCESSFUL in 837ms
+14 actionable tasks: 3 executed, 11 up-to-date
+```
+
+This is automatically run when you build your Android application.
+</TabItem>
+<TabItem value="ios" label="iOS">
+Codegen is run as part of the script phases that's automatically added to the project generated by Cocoapods.
+
+```bash
+cd ios
+bundle install
+bundle exec pod install
+bundle exec pod install
+
+...
+Framework build type is static library
+[Codegen] Adding script_phases to ReactCodegen.
+[Codegen] Generating ./build/generated/ios/ReactCodegen.podspec.json
+[Codegen] Analyzing /Users/me/src/TurboModuleExample/package.json
+[Codegen] Searching for codegen-enabled libraries in the app.
+[Codegen] Found TurboModuleExample
+[Codegen] Searching for codegen-enabled libraries in the project dependencies.
+[Codegen] Found react-native
+...
+```
+
+</TabItem>
+</Tabs>
```

Comment:
I think we have to have these included, otherwise the guide is incomplete.  If I try open these projects up in their respective editors:

## Xcode
Can't until the xcworspace has been generated.

## Android Studio

### Before
![CleanShot 2024-10-17 at 16 18 11@2x](https://github.com/user-attachments/assets/61cd83b5-6c67-4d44-b1db-3ee9bdca7c64)

### After
![CleanShot 2024-10-17 at 16 19 56@2x](https://github.com/user-attachments/assets/feedabf8-b0ff-4d9a-a6a0-5eaebc138bf1)

Reply:
`pod install` needs to be executed for sure, but I would remove the lines that refers to codegen, that's an implementation detail for the user.

I never used Android studio, but I can see the problem. The thing is that we would like for codegen to be transparent... If we ask the users to execute it manually, that's not transparent anymore. :(

cc. @cortinico what do you think about this?